lib/main.js

```diff
@@ -841,6 +843,17 @@ exports.typeset = function (data,callback) {
   if (serverState == STATE.READY) StartQueue();
 }
 
+// main API, callback and promise compatible
+exports.typeset = function (data, callback) {
+    if (callback) cbTypeset(data, callback);
+    else return new Promise(function (resolve, reject) {
+        cbTypeset(data, function (output, input) {
+            if (output.errors) reject(output.errors);
```

Comment:
Do we need to be worried about the fact that `output.errors` an include warnings (like missing characters)?  There is a configuration option for turning this off, so it may not be necessary to worry about that here.